TiDB Cloud: add a tutorial for Looker Studio (#16111)
pingcap/docs: develop/dev-guide-bi-looker-stduio.md

---
title: Connect to TiDB Cloud with Looker Studio
summary: Learn how to connect to TiDB Cloud using Looker Studio
---

# Connect to TiDB Cloud with Looker Studio

TiDB is a MySQL-compatible database, and [Looker Studio](https://lookerstudio.google.com/) is a free web-based BI tool that can visualize data from various sources. 

In this tutorial, you can learn how to connect to your TiDB cluster using Looker Studio.

> **Note:**
>
> - This tutorial is compatible with TiDB Serverless.

## Prerequisites

To complete this tutorial, you need:

> - Google Account that can use Looker Studio.

Reviewer: Is this any Google Account? How do I know if my account can use this?

Author: Yes, every Google Account can use Looker Studio. I will remove "can use Looker Studio" part.

> - A TiDB Serverless cluster.

Reviewer: Doesn't this also work with a dedicated cluster?

Reviewer: Maybe say something like "For this tutorial we will use a serverless cluster. Looker Studio can also be used with dedicated clusters."

Author: @dveeden Thanks, that should work with a dedicated cluster as well. I will check and add TiDB Dedicated Tab.

**If you don't have a TiDB Serverless cluster, you can create one as follows:**

- Follow [Creating a TiDB Serverless cluster](/develop/dev-guide-build-cluster-in-cloud.md) to create your own TiDB Cloud cluster.

## Import Dataset

You can use the S&P 500 dataset provided as an Interactive Tutorial of TiDB Serverless. 

> <SimpleTab>

Reviewer: From CI: `ERROR: develop/dev-guide-bi-looker-stduio.md has unclosed tags: <SimpleTab>.`

4. After the status changed to **Imported**, you can close this dialog. Click the **Exit Tutorial** button in the bottom-left corner of the dialog. 

</div>

## Connect to TiDB

1. Navigate to the [**Clusters**](https://tidbcloud.com/console/clusters) page, and then click the name of your target cluster to go to its overview page.

2. Click **Connect** in the upper-right corner. A connection dialog is displayed.

3. Set **Operating System** to `Windows` in the connection dialog. Then Set **Connect With** to `General`. You can download [CA cert](https://letsencrypt.org/certs/isrgrootx1.pem) at the note below. 

    > **Tip:**
    >
    > As TiDB Serverless requires a secure TLS connection between the client and the cluster, you need this CA cert for connection settings on Looker Studio.

4. Click **Generate Password** to create a random password.

    > **Tip:**
    >
    > If you have created a password before, use the original password or click **Reset Password** to generate a new one.

5. Open Looker Studio and create a Blank Report. Select the' MySQL' connector in the **Add data to report** screen.

> 6. In the setting pane, configure the following connection parameters:

Reviewer 2: ```suggestion
6. In the **BASIC** setting pane, configure the following connection parameters:
```

    - **Host Name or IP**: enter the `HOST` parameter from the TiDB Cloud connection dialog.
    - **Port(Optional)**: enter the `PORT` parameter from the TiDB Cloud connection dialog.
    - **Database**: enter the database you want to connect to. In this tutorial, set `sp500insight`.
    - **Username**: enter the `USERNAME` parameter from the TiDB Cloud connection dialog.
    - **Password**: enter the `PASSWORD` parameter from the TiDB Cloud connection dialog.
    - Check **Enable SSL**.
        - Click the icon on the right of **MySQL SSL Client Configuration Files**. Then, select the downloaded CA file in step 3. 

    ![Looker Studio: configure connection settings for TiDB Serverless](/media/develop/looker-studio-configure-connection.png)

7. Click **AUTHENTICATE**. 

If the authentication succeeds, you can see tables in the database. 

## Create a simple chart

Now, you can use TiDB Serverless as a data source. Let's make a simple chart with data. 

1. In the setting pane, select **CUSTOM QUERY**. 

2. Put the SQL below, then click **Add** in the bottom-right corner. 

>     ```sql

Reviewer: ````suggestion

    ```sql
````

This might fix this:
```
[2024-01-11T07:11:19.119Z] develop/dev-guide-bi-looker-stduio.md: this file has manual line breaks in the following lines:
[2024-01-11T07:11:19.119Z] 
[2024-01-11T07:11:19.119Z] MANUAL LINE BREAKS: L99
```

Reviewer: And this:

```
[2024-01-11T07:11:11.976Z] develop/dev-guide-bi-looker-stduio.md: 87: MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```sql"]
```

    - In the **STYLE** Tab:
      - Series #1 check `Line`.
      - Series #1 axis `Right`.
      - Series #2 check `Bars`.
      - Series #2 axis `Left`.
    - Leave others as defaults.

You can see a combo chart below:

![Looker Studio: A simple Combo Chart](/media/develop/looker-studio-simple-chart.png)

## Next steps

- Learn more usage of Looker Studio from [Looker Studio Help](https://support.google.com/looker-studio)
- Learn the best practices for TiDB application development with the chapters in the [Developer guide](/develop/dev-guide-overview.md), such as [Insert data](/develop/dev-guide-insert-data.md), [Update data](/develop/dev-guide-update-data.md), [Delete data](/develop/dev-guide-delete-data.md), [Single table reading](/develop/dev-guide-get-data-from-single-table.md), [Transactions](/develop/dev-guide-transaction-overview.md), and [SQL performance optimization](/develop/dev-guide-optimize-sql-overview.md).
- Learn through the professional [TiDB developer courses](https://www.pingcap.com/education/) and earn [TiDB certifications](https://www.pingcap.com/education/certification/) after passing the exam.

## Need help?

> Ask questions on [Discord](https://discord.gg/DQZ2dy3cuc?utm_source=doc) or [create a support ticket](https://support.pingcap.com/).

Reviewer: ```suggestion
Ask questions on [Discord](https://discord.gg/DQZ2dy3cuc?utm_source=doc) or [create a support ticket](/support.md).
```

Author: This can't be resolved due to outdated, but it was fixed.

Author: @dveeden This change broke ci/tidb-cloud-check. reverted.
